app/src/main/java/com/bnyro/contacts/util/CalendarUtils.kt

```diff
@@ -22,10 +25,24 @@ object CalendarUtils {
     fun dateToMillis(date: String) = isoDateFormat.parse(date)?.time
 
     fun localizeIsoDate(isoDate: String): String {
+        val noYear = isoDate.startsWith('-')
+        val newIsoDate = if (noYear) isoDate.replace("--", "") else isoDate
+
         val date = runCatching {
-            isoDateFormat.parse(isoDate) ?: return isoDate
-        }.getOrNull() ?: return isoDate
-        return localizedFormat.format(date) ?: isoDate
+            if (noYear) {
+                isoDateFormatWithoutYear.parse(newIsoDate)
+            } else {
+                isoDateFormat.parse(isoDate)
+            } ?: return newIsoDate
+        }.getOrNull() ?: return newIsoDate
+        return localizedFormat.format(date).let {
+            if (noYear) {
+                // Due to the limitations of the API we're getting 1970 when there's no year
+                it.replace("1970", "")
+            } else {
+                it
+            }
+        } ?: newIsoDate
```

Comment:
I don't think anything could be null here (otherwise we should do `localizedFormat(date)?.let {`